### PR TITLE
Deploy a Stellar Asset Contract (SAC) for a Stellar asset using code

### LINKED ISSUE #615: Create a guide for deploying a Stellar Asset Contract for a Stellar asset using code
**Get awarded:** up to $250 worth of XLM
**Section:** How-To Guides - Conventions: https://github.com/stellar/stellar-docs/tree/main/docs/smart-contracts/guides/conventions

**Guide could include:**

- Code walkthrough
- Written explanation

### PR DESCRIPTION
**Pull request by KodeSage**
Bounty PR closes #615

### CODE REVIEW

**Comment by KodeSage** [comment]
@briwylde08 this is ready for review.

**Comment by briwylde08** [comment]
Thank you! We'll get this assigned and reviewed!

File: docs/build/guides/conventions/deploy-sac-with-code.mdx [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+  } catch (error) {
+    console.error("Transaction failed:", error);
+    throw error;
+  }
+}
+function sleep(ms) {
+  return new Promise((resolve) => setTimeout(resolve, ms));
+}
+
+const DeployStellarAssetContract = async () => {
```
Comment (anataliocs) [inline docs/build/guides/conventions/deploy-sac-with-code.mdx:81]:
```suggestion
const deployStellarAssetContract = async () => {
```

File: docs/build/guides/conventions/deploy-sac-with-code.mdx [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+    const feedback = await yeetTx(result, `SACTransactionFeeback`);
+    const contractBuffer = feedback.returnValue._value._value;
+    const contractID = StellarSdk.StrKey.encodeContract(contractBuffer);
+    console.log(`ContractID of Our ${customAsset.code} Asset`, contractID);
+  } catch (e) {
+    console.error("An error occurred while Deploying assets:", e);
+  }
+};
+
+DeployStellarAssetContract();
```
Comment (anataliocs) [inline docs/build/guides/conventions/deploy-sac-with-code.mdx:117]:
```suggestion
await deployStellarAssetContract();
```

File: docs/build/guides/conventions/deploy-sac-with-code.mdx [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+  return new Promise((resolve) => setTimeout(resolve, ms));
+}
+```
+
+This function is a helper that returns a promise that resolves after a specified number of milliseconds. It's used to delay the next check in yeetTx when the transaction status is "NOT_FOUND."
+
+**`DeployStellarAssetContract` function**
+
+```javascript
+const DeployStellarAssetContract = async () => {
```
Comment (anataliocs) [inline docs/build/guides/conventions/deploy-sac-with-code.mdx:195]:
```suggestion
const deployStellarAssetContract = async () => {
```

File: docs/build/guides/conventions/deploy-sac-with-code.mdx [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+    const feedback = await yeetTx(result, `SACTransactionFeeback`);
+    const contractBuffer = feedback.returnValue._value._value;
+    const contractID = StellarSdk.StrKey.encodeContract(contractBuffer);
+    console.log(`ContractID of Our ${customAsset.code} Asset`, contractID);
+  } catch (e) {
+    console.error("An error occurred while Deploying assets:", e);
+  }
+};
+
+DeployStellarAssetContract();
```
Comment (anataliocs) [inline docs/build/guides/conventions/deploy-sac-with-code.mdx:232]:
```suggestion
await deployStellarAssetContract();
```

**Review by anataliocs — CHANGES_REQUESTED** [review]

**Comment by anataliocs** [comment]
Stellar Asset Contract deployed as expected

image


https://stellar.expert/explorer/testnet/contract/CBF2X6JDA5BRMITM5VIHACY6YVWX2NRJQ7YN7EN3KXHXWWX66Z3DSMFU
image

File: docs/build/guides/conventions/deploy-sac-with-code.mdx [hunk 5]
```diff
@@ -0,0 +1,242 @@
...
+## Overview
+
+In this guide, you'll learn how to deploy a [Stellar Asset Contract (SAC)](../../../tokens/stellar-asset-contract.mdx) for a Stellar asset using the [Stellar SDK](../../../tools/sdks/library.mdx#javascript-sdk). The Stellar SDK is a set of tools and library designed to help developers build applications that interact with the Stellar blockchain network.
+
+## Prerequisites:
+
+- [Node.js ](https://nodejs.org/en) and npm installed.
+- Stellar SDK for [JavaScript](https://www.npmjs.com/package/@stellar/stellar-sdk) installed
+- [Knowledge about Issuing an Asset on Stellar](https://developers.stellar.org/docs/tokens/how-to-issue-an-asset)
+
```
[the remark this answers is not in the document]
Comment (KodeSage) [reply docs/build/guides/conventions/deploy-sac-with-code.mdx:30]: Hi @anataliocs do you mean `.env` stuff here?

**Comment by KodeSage** [comment]
Hi @anataliocs , Thank you for the feedbacks.
I pushed up the fixes, which includes
- nesting of sections
- editorial fixes

**Comment by KodeSage** [comment]
Hello @briwylde08 @anataliocs , just checking in for the status of the review.

**Comment by briwylde08** [comment]
Hi @KodeSage! Sorry for the delay! @anataliocs is out of town but will get to the re-review asap!

**Comment by KodeSage** [comment]
HI @briwylde08 @anataliocs , just checking in again for the status of the review.

**Comment by briwylde08** [comment]
Hi @KodeSage!! Sorry for the delay here, will get this reviewed asap. Thank you for the flag!

**Comment by KodeSage** [comment]
> Hi @KodeSage!! Sorry for the delay here, will get this reviewed asap. Thank you for the flag!

alright thank you @briwylde08

**Review by anataliocs — APPROVED** [review]
Checked out changes locally!  LGTM 🚢 🚢 🚢 

image

**Comment by KodeSage** [comment]
> Checked out changes locally! LGTM 🚢 🚢 🚢
> 
> image

Thank you @anataliocs 🚀
Hi @briwylde08 this has been approved.

**Comment by briwylde08** [comment]
Thanks again for your contribution @KodeSage! We'll take the award conversation back to the corresponding issue.

**Pull request merged**